ThunderCloud/AuthenticationController.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  AuthenticationController.swift
+//  ThunderCloud
+//
+//  Created by Simon Mitchell on 15/02/2018.
+//  Copyright © 2018 threesidedcube. All rights reserved.
+//
+
+import Foundation
+import ThunderRequest
+
+/// The Authentication Controller is responsible for authenticating with the
+/// Storm CMS API. Any endpoint that requires an authorisation token can use
+/// this controller to obtain a token.
+public class AuthenticationController {
+	
+	/// A structual representation of the user's authentication
+	public struct Authorization: Decodable, Encodable {
+		
+		/// The authorization token
+		let token: String
+		
+		/// The expiry date of the Authorization
+		let expiry: Date
+		
+		/// Whether the user is currently authenticated
+		var hasExpired: Bool {
+			return expiry < Date()
+		}
+	}
+	
+	/// A typealias for an authentication completion
+	public typealias AuthenticationCompletion = (_ authorization: Authorization?, _ error: Error?) -> Void
+	
+	private let requestController: TSCRequestController
+	
+	public init() {
+		
+		requestController = TSCRequestController(baseAddress: "https://auth.cubeapis.com/v1.5")
+	}
+	
+	public func authenticateWith(username: String, password: String, completion: AuthenticationCompletion?) {
+		
+		requestController.post("authentication", bodyParams: ["username": username, "password": password]) { (response, error) in
+			
+			guard error == nil else {
```

Comment:
This is exactly why I like `guard` so much, such a clean flow! Very nice indeed :)